From 8fc43d907d34ee0ae308b800422e8268fcb30316 Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@intel.com>
Date: Tue, 15 Jul 2014 14:16:07 +0200
Subject: asoc: intel: update bias_level on card context only

Don't set bias_level in init. snd_soc_instantiate_card has already
initialized it.

Issue: GMIN-1544
Change-Id: Iec63a770bac623666f686e9dd82c95b4117a346c
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@intel.com>
---
 sound/soc/intel/board/byt_cr_dpcm_rt5640.c | 5 ++---
 sound/soc/intel/board/byt_cr_dpcm_rt5651.c | 6 ++----
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
index a9db98b..7bbc8bb 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
@@ -653,7 +653,8 @@ static int byt_set_bias_level(struct snd_soc_card *card,
 		pr_err("%s: Invalid bias level=%d\n", __func__, level);
 		return -EINVAL;
 	}
-	card->dapm.bias_level = level;
+	if (&card->dapm == dapm)
+		card->dapm.bias_level = level;
 	pr_debug("card(%s)->bias_level %u\n", card->name,
 			card->dapm.bias_level);
 	return 0;
@@ -674,8 +675,6 @@ static int byt_init(struct snd_soc_pcm_runtime *runtime)
 		return -EIO;
 	}
 
-	/* Set codec bias level */
-	byt_set_bias_level(card, &card->dapm, SND_SOC_BIAS_OFF);
 	card->dapm.idle_bias_off = true;
 
 	/* Threshold base = 2000uA; scale factor = 0.5 =>
diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5651.c b/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
index 6e002dd..680b195 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
@@ -541,7 +541,8 @@ static int byt_set_bias_level(struct snd_soc_card *card,
 		return -EINVAL;
 	}
 
-	card->dapm.bias_level = level;
+	if (&card->dapm == dapm)
+		card->dapm.bias_level = level;
 	pr_debug("card(%s)->bias_level %u\n", card->name,
 			card->dapm.bias_level);
 	return 0;
@@ -551,7 +552,6 @@ static int byt_init(struct snd_soc_pcm_runtime *runtime)
 {
 	int ret;
 	struct snd_soc_codec *codec;
-	struct snd_soc_dapm_context *dapm = &codec->dapm;
 	struct snd_soc_card *card = runtime->card;
 	struct byt_mc_private *ctx = snd_soc_card_get_drvdata(runtime->card);
 	int codec_gpio, jd_gpio;
@@ -574,8 +574,6 @@ static int byt_init(struct snd_soc_pcm_runtime *runtime)
 	pr_info("%s: JD GPIO = %d", __func__, jd_gpio);
 	hs_gpio[0].gpio = jd_gpio;
 
-	/* Set codec bias level */
-	byt_set_bias_level(card, dapm, SND_SOC_BIAS_OFF);
 	card->dapm.idle_bias_off = true;
 	/* Set overcurrent detection threshold base and scale factor
 	   for jack type identification and button events. */
-- 
2.0.1

